hack/test-go.sh

```diff
@@ -256,18 +256,13 @@ runTests() {
   # must make sure the output from parallel runs is not mixed. To achieve this,
   # we spawn a subshell for each parallel process, redirecting the output to
   # separate files.
-  printf "%s\n" "${@}" | xargs -I{} -n1 -P${KUBE_COVERPROCS} \
-    bash -c "set -o pipefail; _pkg=\"{}\"; _pkg_out=\${_pkg//\//_}; \
-        go test ${goflags[@]:+${goflags[@]}} \
-          ${KUBE_RACE} \
-          ${KUBE_TIMEOUT} \
-          -cover -covermode=\"${KUBE_COVERMODE}\" \
-          -coverprofile=\"${cover_report_dir}/\${_pkg}/${cover_profile}\" \
-          \"${KUBE_GO_PACKAGE}/\${_pkg}\" \
-          ${testargs[@]:+${testargs[@]}} \
-        | tee ${junit_filename_prefix:+\"${junit_filename_prefix}-\$_pkg_out.stdout\"} \
-        | grep \"${go_test_grep_pattern}\"" \
-      && test_result=$? || test_result=$?
+
```

Comment:
@ixdy I changed the code but the UT on GNU did not pass, if I run `KUBE_COVER=y hack/test-go.sh` and `echo ${@}` here, I can see all directories are listed, but only one is passed to run-go-test.sh. Can you help me to take a look and show some hints? thanks.

In addition, as BSD accepts -p as pattern for grep, while it is -e on GNU, to make the code same on both platforms, I remove the -p or -e, so go_test_grep_pattern has the same problem with original version for this commit.